File: algorithms/sorting/selectionSort.py
"""
Algorithm for selection sort
"""

def selection_sort(arr):

    for i in range(len(arr)):
        min_index = i

        for j in range(i+1, len(arr)):

            if arr[min_index] > arr[j]:

                min_index = j

        arr[min_index], arr[i] = arr[i], arr[min_index]

    return arr


print(selection_sort([4,2,6,7,8,12,5,8,19]))

File: algorithms/CPP/Sorting/selectionSort.cpp
#include<iostream>
using namespace std;

// Swap two numbers
void swap(int *xp, int *yp){
  int temp = *xp;
  *xp = *yp;
  *yp = temp;
}
// Selection sort algorithm
void selectionSort(int arr[], int n){
  int minSoFar;
  for(int i=0; i<n; i++){
    minSoFar = i;
    for(int j = i+1; j<n; j++){
      if(arr[minSoFar] > arr[j]){
        minSoFar = j;
      }
    }
    if(minSoFar!= i){
      swap(&arr[i], &arr[minSoFar]);
    }
  }
}

// Traversing the array
void printArray(int arr[], int size){
  for(int i = 0; i < size; i++){
    cout << arr[i] << " ";
  }
  cout << " " << endl;
}

// main function
int main(){
  int arr[] = {10,9,8,7,6,5,4,3,2,1};
  int n = sizeof(arr)/sizeof(arr[0]);
  selectionSort(arr,n);
  printf("Sorted Array using bubble sort is ");
  printArray(arr, n);
  return 0;
}
